MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1887,14 +1898,22 @@ public Person findBestInRole(int role, String skill) {
         }
         // also need to loop through and collect engineers on self-crewed vessels
         for (Unit u : getUnits()) {
-            if (u.isSelfCrewed() && !(u.getEntity() instanceof Infantry) && null != u.getEngineer()) {
+            if (u.isSelfCrewed() && !(u.getEntity() instanceof Infantry) && (null != u.getEngineer())) {
                 techs.add(u.getEngineer());
             }
         }
 
         // Return the tech collection sorted worst to best
         // Reverse the sort if we've been asked for best to worst
         if (sorted) {
+            // First order by the amount of time the person has remaining, with the highest amount
+            // of time being at the top of the list
+            if (eliteFirst) {
+                techs.sort(Comparator.comparingInt(Person::getMinutesLeft));
```

Comment:
How is this "elite first"?

Reply:
It works in practice, but I can't for the life of me understand why.

Reply:
This doesn't sort elites first. The code which does is on line 1934:
```
                // Return, swapping the value if we're looking to have Elites ordered first
                return eliteFirst ? -retVal : retVal;
```
I think the logic should be revised for this block, as it stands it doesn't make much sense (even if it produces the desired results).

Reply:
I've added comments which should make it easier to understand.

Reply:
I guess I don't follow. If you *don't* want elites first, why wouldn't you still want the person with the most time left?

Reply:
Yes, this is used for Mothballing, which you always want a person with the most time left sorted to the top while the best personnel are the worst skilled. Annoying logic, but it works and looks good in practice.